ci/jenkins/test.sh

```diff
@@ -474,7 +490,11 @@ function run_e2e {
     mkdir -p `pwd`/antrea-test-logs
     # HACK: see https://github.com/antrea-io/antrea/issues/2292
     go mod edit -replace github.com/moby/spdystream=github.com/antoninbas/spdystream@v0.2.1 && go mod tidy
-    go test -v antrea.io/antrea/test/e2e --logs-export-dir `pwd`/antrea-test-logs --provider remote -timeout=100m --prometheus
+    if [[ $FLEXIBLE_IPAM == true ]]; then
+        go test -v antrea.io/antrea/test/e2e --logs-export-dir `pwd`/antrea-test-logs --provider remote -timeout=100m --prometheus --antrea-ipam
```

Comment:
I wasn't aware that the e2e tests needed to be aware of the IPAM mode. That seems unnecessary to me, and I feel like we should be running the same tests, but maybe I'm missing something. cc @gran-vmv

Reply:
@antoninbas For system requirements, AntreaIPAM can be enabled only in this new testbed. 
And AntreaIPAM testbed will run a special set of e2e cases, which including most of existing cases with AntreaIPAM enabled, and some new e2e cases related to AntreaIPAM Pods.